Course_Part_08/src/assignment30/Shape.java

```diff
@@ -0,0 +1,33 @@
+package assignment30;
+
+public abstract class Shape {
+
+    private String name;
+
+    public Shape(String name) {
+        this.name = name;
+    }
+
+    public abstract void draw();
+
+    public abstract double calculateArea();
+
+    public abstract double calculatePerimeter();
+
+    public double check_value(double side) {
```

Comment:
methode namen schrijven we binnen Java vrijwel nooit met een underscore tussen woorden.

```
checkValue(double side)
```